app/src/main/java/com/tari/android/wallet/infrastructure/security/encryption/SymmetricEncryptionAlgorithm.kt

```diff
@@ -0,0 +1,56 @@
+package com.tari.android.wallet.infrastructure.security.encryption
+
+import java.io.InputStream
+import java.io.OutputStream
+import java.security.MessageDigest
+import javax.crypto.Cipher
+import javax.crypto.CipherOutputStream
+import javax.crypto.spec.SecretKeySpec
+
+interface SymmetricEncryptionAlgorithm {
+
+    fun encrypt(key: CharArray, isProvider: () -> InputStream, osProvider: () -> OutputStream)
+
+    companion object {
+        fun aes(): SymmetricEncryptionAlgorithm = AdvancedEncryptionStandard()
+    }
+
+}
+
+private class AdvancedEncryptionStandard(
```

Comment:
I think you can safely go for `AES` for a name here, [Apple's API Design Guideline](https://swift.org/documentation/api-design-guidelines/) has a cool note on abbreviations which could apply:

>The intended meaning for any abbreviation you use should be easily found by a web search.

Reply:
most of the time when I used this term I got a question "what is AES", that's why I didn't used an acronym here

it will be changed based on comment though

Reply:
Ye "what is AES" gives spot on result on Google, so I thought it was safe.